lib/pure/strutils.nim

```diff
@@ -2813,6 +2813,58 @@ proc format*(formatstr: string, a: varargs[string, `$`]): string {.noSideEffect,
   result = newStringOfCap(formatstr.len + a.len)
   addf(result, formatstr, a)
 
+proc stripInplace*(s: var string, leading = true, trailing = true,
+                   chars: set[char] = Whitespace) {.since: (1, 5, 1).} =
+  ## Inplace version of `strip`. Strips leading or 
+  ## trailing `chars` (default: whitespace characters).
+  ##
+  ## If `leading` is true (default), leading `chars` are stripped.
+  ## If `trailing` is true (default), trailing `chars` are stripped.
+  ## If both are false, the string is unchanged.
+  ##
+  ## See also:
+  ## * `strip proc<#strip,string,set[char]>`_
+  runnableExamples:
+    var a = "  vhellov   "
+    stripInplace(a)
+    doAssert a == "vhellov"
+
+    a = "  vhellov   "
+    a.stripInplace(leading = false)
+    doAssert a == "  vhellov"
+
+    a = "  vhellov   "
+    a.stripInplace(trailing = false)
+    doAssert a == "vhellov   "
+
+    var c = "blaXbla"
+    c.stripInplace(chars = {'b', 'a'})
+    doAssert c == "laXbl"
+    c = "blaXbla"
+    c.stripInplace(chars = {'b', 'a', 'l'})
+    doAssert c == "X"
+
+  var
+    first = 0
+    last = high(s)
+  if leading:
+    while first <= last and s[first] in chars: inc(first)
+  if trailing:
+    while last >= 0 and s[last] in chars: dec(last)
+
+  if first > 0:
+    when nimvm:
+      for index in first .. last:
```

Comment:
use this DRY pattern instead via an auxiliary template, which I use in all similar situations:
```nim
template fn = 
  for index in first .. last:
    s[index - first] = s[index]

when nimvm: fn()
else:
  when not declared(moveMem): fn()
  else: moveMem(addr s[0], addr s[first], last - first + 1)
```

Reply:
nit: I really prefer ``template impl`` over ``template fn``.